Kubernetes/Helm workspaceVolume declaration in values.yaml results in "can't evaluate field type in type interface".

**Describe the bug**
Defining a workspace volume declaration in a copy of the default values.yaml results in <.Values.agent.workspaceVolume.type>: can't evaluate field type in type interface {}


**Version of Helm and Kubernetes**:


```console
$ helm version
helm version
version.BuildInfo{Version:"v3.7.2", GitCommit:"663a896f4a815053445eec4153677ddc24a0a361", GitTreeState:"clean", GoVersion:"go1.16.10"}
```

Kubernetes Version:

```console
$ kubectl version
kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:41:01Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.7", GitCommit:"1f86634ff08f37e54e8bfcd86bc90b61c98f84d4", GitTreeState:"clean", BuildDate:"2021-11-17T14:35:38Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.23) and server (1.21) exceeds the supported minor version skew of +/-1
```


**Which version of the chart**:
Default

**What happened**:
```
Replicated both with my customized values and a values.yaml file with everything else being default:

values.yaml:
  # You can define the workspaceVolume that you want to mount for this container
  # Allowed types are: DynamicPVC, EmptyDir, HostPath, Nfs, PVC
  # Configure the attributes as they appear in the corresponding Java class for that type
  # https://github.com/jenkinsci/kubernetes-plugin/tree/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/volumes/workspace
  workspaceVolume:
  # - type: DynamicPVC
  #   configMapName: myconfigmap
  # - type: EmptyDir
  #   memory: false
  # - type: HostPath
  #   hostPath: /var/lib/containers
   - type: Nfs
     readOnly: false
     serverAddress: "192.0.2.0"
     serverPath: /var/lib/containers



Installation w/ just the problematic workspace volume setting:

helm install testingjenkins jenkinsci/jenkins -f newvalues.yaml
coalesce.go:199: warning: cannot overwrite table with non table for workspaceVolume (map[])
Error: INSTALLATION FAILED: template: jenkins/templates/jcasc-config.yaml:43:8: executing "jenkins/templates/jcasc-config.yaml" at <include "jenkins.casc.defaults" .>: error calling include: template: jenkins/templates/_helpers.tpl:164:10: executing "jenkins.casc.defaults" at <include "jenkins.casc.podTemplate" .>: error calling include: template: jenkins/templates/_helpers.tpl:292:22: executing "jenkins.casc.podTemplate" at <.Values.agent.workspaceVolume.type>: can't evaluate field type in type interface {}
```

**What you expected to happen**:

```
A successful installation much like this one was. This one was WITHOUT the needed workspaceVolume setting.

helm install --namespace jenkins jenkins jenkinsci/jenkins -f values.yaml
NAME: jenkins
LAST DEPLOYED: Sun Jan 16 14:20:17 2022
NAMESPACE: jenkins
STATUS: deployed
REVISION: 1
NOTES:
1. Get your 'admin' user password by running:
  kubectl exec --namespace jenkins -it svc/jenkins -c jenkins -- /bin/cat /run/secrets/chart-admin-password && echo
2. Get the Jenkins URL to visit by running these commands in the same shell:
  echo http://127.0.0.1:8080
  kubectl --namespace jenkins port-forward svc/jenkins 8080:8080

3. Login with the password from step 1 and the username: admin
4. Configure security realm and authorization strategy
5. Use Jenkins Configuration as Code by specifying configScripts in your values.yaml file, see documentation: http:///configuration-as-code and examples: https://github.com/jenkinsci/configuration-as-code-plugin/tree/master/demos

For more information on running Jenkins on Kubernetes, visit:
https://cloud.google.com/solutions/jenkins-on-container-engine

For more information about Jenkins Configuration as Code, visit:
https://jenkins.io/projects/jcasc/


NOTE: Consider using a custom image with pre-installed plugins
```

**How to reproduce it** (as minimally and precisely as possible):

1. Have the helm repo installed.
2. edit values.yaml to specify a workspaceVolume.
2. helm install jenkins jenkinsci/jenkins -f values.yaml
3. It does not work.


**Other info**

This is a RKE-based on-premise Rancher install. I've attempted the workspaceVolume mount w/ both NFS and PVC claims, both give similar errors. 

I've attached my example newvalues.yaml
[newvalues.zip](https://github.com/jenkinsci/helm-charts/files/7877942/newvalues.zip)
.

--- a/charts/jenkins/CHANGELOG.md
+++ b/charts/jenkins/CHANGELOG.md
@@ -12,6 +12,10 @@ Use the following links to reference issues, PRs, and commits prior to v2.6.0.
 The change log until v1.5.7 was auto-generated based on git commits.
 Those entries include a reference to the git commit to be able to get more details.
 
+## 3.11.2
+
+Improve example for workspaceVolume. Clarify that this is not a list.
+
 ## 3.11.1
 
 Update configuration-as-code plugin to 1.55.1

--- a/charts/jenkins/Chart.yaml
+++ b/charts/jenkins/Chart.yaml
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: jenkins
 home: https://jenkins.io/
-version: 3.11.1
+version: 3.11.2
 appVersion: 2.319.2
 description: Jenkins - Build great things at any scale! The leading open source automation server, Jenkins provides hundreds of plugins to support building, deploying and automating any project.
 sources:

--- a/charts/jenkins/values.yaml
+++ b/charts/jenkins/values.yaml
@@ -648,19 +648,25 @@ agent:
   # Configure the attributes as they appear in the corresponding Java class for that type
   # https://github.com/jenkinsci/kubernetes-plugin/tree/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/volumes/workspace
   workspaceVolume: {}
-  # - type: DynamicPVC
-  #   configMapName: myconfigmap
-  # - type: EmptyDir
-  #   memory: false
-  # - type: HostPath
-  #   hostPath: /var/lib/containers
-  # - type: Nfs
-  #   readOnly: false
-  #   serverAddress: "192.0.2.0"
-  #   serverPath: /var/lib/containers
-  # - type: PVC
-  #   claimName: mypvc
-  #   readOnly: false
+  ## DynamicPVC example
+  # type: DynamicPVC
+  # configMapName: myconfigmap
+  ## EmptyDir example
+  # type: EmptyDir
+  # memory: false
+  ## HostPath example
+  # type: HostPath
+  # hostPath: /var/lib/containers
+  ## NFS example
+  # type: Nfs
+  # readOnly: false
+  # serverAddress: "192.0.2.0"
+  # serverPath: /var/lib/containers
+  ## PVC example
+  # type: PVC
+  # claimName: mypvc
+  # readOnly: false
+  #
   # Pod-wide environment, these vars are visible to any container in the agent pod
   envVars: []
   # - name: PATH

